pkg/service/service.go

```diff
@@ -832,11 +832,21 @@ func (s *Service) upsertService(params *lb.SVC) (bool, lb.ID, error) {
 
 	backendsCopy := []*lb.Backend{}
 	for _, b := range params.Backends {
-		// Local redirect services or services with trafficPolicy=Local may
-		// only use node-local backends for external scope. We implement this by
-		// filtering out all backend IPs which are not a local endpoint.
-		if filterBackends && len(b.NodeName) > 0 && b.NodeName != nodeTypes.GetName() {
-			continue
+		// Local redirect services or services with externalTrafficPolicy=Local
```

Comment:
It's not clear to me how this is intended to interact with `external / internalTrafficPolicy`. Does `delegate-if-local` require a `xTrafficPolicy`? What happens if only external or internal are set, but not both?

`delegate-if-local` implies that delegation should take place if there is a local backend, otherwise don't delegate. That doesn't match `xTrafficPolicy`, which **drops** traffic if there are no local backends.

Reply:
The behavior is as with any other regular service. By default (Cluster) we push to both local & remote backends if present, with Local we only filter to local ones and if none is local then traffic is dropped.